packages/react-native-performance/src/performance.js

```diff
@@ -0,0 +1,213 @@
+import { createEventEmitter } from './event-emitter';
+import { createPerformanceObserver } from './performance-observer';
+import {
+  PerformanceMark,
+  PerformanceMeasure,
+  PerformanceMetric,
+} from './performance-entry';
+
+export const now = () => global.nativePerformanceNow();
+
+export const createPerformance = () => {
+  const timeOrigin = now();
+  const { addEventListener, removeEventListener, emit } = createEventEmitter();
+  const marks = new Map();
+  let entries = [];
+
+  function addEntry(entry) {
+    entries.push(entry);
+    if (entry.entryType === 'mark' || entry.entryType === 'react-native-mark') {
+      marks.set(entry.name, entry.startTime);
+    }
+    emit(entry);
+    return entry;
+  }
+
+  const removeEntries = (type, name) => {
+    entries = entries.filter(entry => {
+      if (entry.entryType === type && (!name || entry.name === name)) {
+        marks.delete(entry.name);
+        return false;
+      }
+      return true;
+    });
+  };
+
+  const mark = (markName, markOptions = {}) =>
+    addEntry(
+      new PerformanceMark(markName, {
+        startTime:
+          'startTime' in markOptions && markOptions.startTime !== undefined
+            ? markOptions.startTime
+            : now(),
+        detail: markOptions.detail,
+      })
+    );
+
+  const clearMarks = name => removeEntries('mark', name);
+
+  const clearMeasures = name => removeEntries('measure', name);
+
+  const clearMetrics = name => removeEntries('metric', name);
+
+  const convertMarkToTimestamp = markOrTimestamp => {
+    switch (typeof markOrTimestamp) {
+      case 'string': {
+        if (!marks.has(markOrTimestamp)) {
+          throw new Error(
```

Comment:
Isn't throwing an error too much for a logger? Maybe it can fail silently. It is valid for all measure events.

Reply:
This is consistent with the Chrome implementation and the [spec](https://www.w3.org/TR/user-timing-3/#dfn-convert-a-mark-to-a-timestamp).

> If no matching entry is found, throw a SyntaxError